device_doctor/lib/src/android_device.dart

```diff
@@ -249,6 +256,51 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
     return healthCheckResult;
   }
 
+  /// The health check for battery level.
+  Future<HealthCheckResult> batteryLevelCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
```

Comment:
Any special reason to expect an exception? the code is just querying a property.

Reply:
There used to be an error exception when calling `adb`, and this is supposed to catch that.